[DM-9847] Add command line interface for Flake8, TravisCI and GitHub …

jmatt opened the pull request:
…branch protection.

	modified:   .gitignore
	new file:   Makefile
	new file:   README.md
	new file:   files/.travis.yml
	new file:   files/repo_names.txt
	new file:   files/setup.cfg
	new file:   github-travis-flake8.py
	new file:   repos/.gitignore

jmatt committed:
[DM-9847] Add command line interface for Flake8, TravisCI and GitHub branch protection.

	modified:   .gitignore
	new file:   Makefile
	new file:   README.md
	new file:   files/.travis.yml
	new file:   files/repo_names.txt
	new file:   files/setup.cfg
	new file:   github-travis-flake8.py
	new file:   repos/.gitignore

jmatt committed:
Refactor into three seperate CLI scripts and one that has all functionality.

	renamed:    github-travis-flake8.py -> gtf/base.py
	new file:   gtf/github-protect-travis.py
	new file:   gtf/protect.py
	new file:   gtf/travis.py
	new file:   gtf/update.py

athornton commented on files/repo_names.txt:
```diff
@@ -0,0 +1,32 @@
+sqre-codekit
```
Do we maybe want to generate this list on the fly by asking GitHub instead?  Maybe not, since it probably isn't all repos.  Hm.

athornton reviewed (approved):
Generally looks good.  I'm with Jonathan, though, I think for not-C++ stuff, 80 columns and no flake8 ignoring is a better idea.  But I'm not religious about it.

jmatt committed:
Refactor and remove redundant code.

  * Remove `#!/usr/bin/env python` from base.py.
  * Remove unused methods and imports.
  * PEP8 Compliance.

jmatt committed:
Remove extraneous #! line.

	modified:   base.py

jmatt committed:
sqre-gtf 0.1.2. Add to pypi, new task strucutre and cli.

  * Add tasks strucuture. To add a new task simplying implement the task method
    in a new module in `gtf.tasks`.

```python
def task(github_repo, git_repo, branch_name, clone_dir, verbose=False):
```

  * Further separate CLI functionality to `gtf.cli`.
  * Add setup.py and pypi as `sqre-gtf`.

	modified:   .gitignore
	renamed:    files/.travis.yml -> .travis.yml
	deleted:    Makefile
	new file:   files/default-.travis.yml
	renamed:    files/repo_names.txt -> files/dm-9847-repo_names.txt
	renamed:    files/setup.cfg -> files/stack-setup.cfg
	modified:   gtf/base.py
	new file:   gtf/cli/__init__.py
	renamed:    gtf/github-protect-travis.py -> gtf/cli/github_protect_travis.py
	new file:   gtf/cli/protect.py
	new file:   gtf/cli/travis.py
	new file:   gtf/cli/update.py
	modified:   gtf/protect.py
	new file:   gtf/tasks/__init__.py
	new file:   gtf/tasks/sqre.py
	new file:   gtf/tasks/stack.py
	new file:   gtf/tasks/test_verbose.py
	modified:   gtf/travis.py
	modified:   gtf/update.py
	deleted:    repos/.gitignore
	new file:   requirements.txt
	new file:   setup.py
